openpype/pipeline/colorspace.py

```diff
@@ -0,0 +1,468 @@
+                                     Defaults to None.
+        project_settings (dict, optional): project settings. Defaults to None.
+        validate (bool, optional): should resulting colorspace be validated
+                                   with config file? Defaults to True.
+
+    Returns:
+        str: name of colorspace
+    """
+    if not any([config_data, file_rules]):
+        project_settings = project_settings or get_project_settings(
+            project_name
+        )
+        config_data = get_imageio_config(
+            project_name, host_name, project_settings)
+        file_rules = get_imageio_file_rules(
+            project_name, host_name, project_settings)
+
+    # match file rule from path
+    colorspace_name = None
+    for _frule_name, file_rule in file_rules.items():
+        pattern = file_rule["pattern"]
+        extension = file_rule["ext"]
+        ext_match = re.match(
+            r".*(?=.{})".format(extension), path
+        )
+        file_match = re.search(
+            pattern, path
+        )
+
+        if ext_match and file_match:
+            colorspace_name = file_rule["colorspace"]
+
+    if not colorspace_name:
+        log.info("No imageio file rule matched input path: '{}'".format(
+            path
+        ))
+        return None
+
+    # validate matching colorspace with config
+    if validate and config_data:
+        validate_imageio_colorspace_in_config(
+            config_data["path"], colorspace_name)
+
+    return colorspace_name
+
+
+def parse_colorspace_from_filepath(
+    path, host_name, project_name,
+    config_data=None,
+    project_settings=None
+):
+    """Parse colorspace name from filepath
+
+    An input path can have colorspace name used as part of name
+    or as folder name.
+
+    Args:
+        path (str): path string
+        host_name (str): host name
+        project_name (str): project name
+        config_data (dict, optional): config path and template in dict.
+                                      Defaults to None.
+        project_settings (dict, optional): project settings. Defaults to None.
+
+    Returns:
+        str: name of colorspace
+    """
+    if not config_data:
+        project_settings = project_settings or get_project_settings(
+            project_name
+        )
+        config_data = get_imageio_config(
+            project_name, host_name, project_settings)
+
+    config_path = config_data["path"]
+
+    # match file rule from path
+    colorspace_name = None
+    colorspaces = get_ocio_config_colorspaces(config_path)
+    for colorspace_key in colorspaces:
+        # check underscored variant of colorspace name
+        # since we are reformating it in integrate.py
+        if colorspace_key.replace(" ", "_") in path:
+            colorspace_name = colorspace_key
+            break
+        if colorspace_key in path:
+            colorspace_name = colorspace_key
+            break
```

Comment:
This seems like quite an unreliable parsing method. I feel like this should either use file rules or don't try this at all - or the logic should be made way more explicit.

A filepath with `aces` or `raw` anywhere in the path will match, but there might be other colorspaces that contain `aces` but are longer?

Some examples of those conflicts could be:
```
acesproxy
acesproxy_ap1
```
```
acescct
acescct_ap1
```
```
rlf_rc
rlf_rc2
rlf_rc3
rlf_rc4
```
```
out_rec709
out_rec709d60sim
```
```
Input - Sony - Linear - S-Gamut
Input - Sony - Linear - S-Gamut Daylight
```
```
Input - Sony - Linear - S-Gamut3
Input - Sony - Linear - S-Gamut3.Cine
```

I've just done a quick search through OCIO 1.0.3 where it could fail to find the right colorspace.

There's a couple of things we could try to do to perform better matching.

- Match only the filename, not the full path
- Take the LONGEST match that is found
- Or be even more explicit and e.g. ALWAYS expect an underscore before the colorspace or be in a certain location of the fname (e.g. before the extension, or before a dot, etc.). This is likely what'll make it most explicit.

Reply:
> - Match only the filename, not the full path
> - Take the LONGEST match that is found
> - Or be even more explicit and e.g. ALWAYS expect an underscore before the colorspace or be in a certain location of the fname (e.g. before the extension, or before a dot, etc.). This is likely what'll make it most explicit.

Yes that is very good point. Since it would be practical to keep full path, because some productions will wont to keep colorspace at folder level - and it is almost impossible to parse only part of name - we have only option to evaluate the longest match. 

I have add it into the next issue. I would address it in next patch release or do you suggest this feature will not be complete without this?

Reply:
I'm not too bothered it being broken for now - the parsing will be somewhat unreliable no matter how far we get. :)

Reply:
Well, we could at least try, but it will never be optimal for all, yeah.. Thanks man!